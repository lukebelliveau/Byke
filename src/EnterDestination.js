// @flow
import React, { Component } from 'react';
import { View, TextInput, StyleSheet, Platform } from 'react-native';

const initialState = {
  enteredText: '',
};

class EnterDestination extends Component {
  state = initialState;
  constructor(props) {
    super(props);
    this.props = props;
  }

  textChanged = event => {
    this.setState({ enteredText: event });
  };

  render() {
    return (
      <View style={[this.props.style, styles.container]}>
        <TextInput
          style={styles.input}
          underlineColorAndroid="transparent"
          placeholder="Where are you going?"
          placeholderTextColor="lightgray"
          onChangeText={this.textChanged}
          onSubmitEditing={() => this.props.onSubmit(this.state.enteredText)}
          returnKeyType="go"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'dodgerblue',
    flexDirection: 'column-reverse',
    alignItems: 'center',
    shadowColor: 'black',
    shadowRadius: 10,
    shadowOpacity: 100,
    elevation: 10,
  },
  input: {
    flex: Platform.OS === 'ios' ? 1 : 0,
    maxHeight: Platform.OS === 'ios' ? 25 : 50,
    fontSize: 15,
    width: '75%',
    backgroundColor: 'deepskyblue',
    color: 'white',
    borderRadius: 10,
    margin: 20,
    padding: 5,
    paddingLeft: 10,
  },
});

export default EnterDestination;
