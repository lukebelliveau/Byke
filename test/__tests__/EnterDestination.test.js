import React from 'react';
import renderer from 'react-test-renderer';
import EnterDestination from '../../src/EnterDestination';

it('should render', () => {
  const input = renderer.create(<EnterDestination/>);

  expect(input).toMatchSnapshot();
})