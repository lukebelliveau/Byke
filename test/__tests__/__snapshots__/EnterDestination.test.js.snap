// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "alignItems": "center",
        "backgroundColor": "dodgerblue",
        "elevation": 10,
        "flexDirection": "column-reverse",
        "shadowColor": "black",
        "shadowOpacity": 100,
        "shadowRadius": 10,
      },
    ]
  }
>
  <TextInput
    placeholder="Where are you going?"
    placeholderTextColor="lightgray"
    style={
      Object {
        "backgroundColor": "deepskyblue",
        "borderRadius": 10,
        "color": "white",
        "flex": 1,
        "fontSize": 15,
        "margin": 20,
        "maxHeight": 25,
        "padding": 5,
        "paddingLeft": 10,
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
